tests/clar.c

```diff
@@ -193,21 +204,33 @@ void cl_trace_register(cl_trace_cb *cb, void *payload)
 
 /* Core test functions */
 static void
-clar_report_errors(void)
+clar_report(int *i, struct clar_error *error)
```

Comment:
By the way, this interface is a bit funny, as it's kind of emulating an iterator via this integer. But in my opinion, using this integer makes code less understandable compared to simply iterating over the linked list of errors ourselves.

Feel free to ignore this comment -- it's not that important after all and can easily be cleaned up later

Reply:
Huh.  Thanks for the detailed review, I had missed that.  I agree and will fix it up before I merge it.

Reply:
Fixed!

Reply:
And, actually, I decided that I didn't like the way that I fixed it.  One of the report functions was still assuming knowledge of the clar structure that a caller also had knowledge of, so I simply passed that in as an arg and I think it's quite a bit nicer.  Thanks again for catching this.